src/test/datascience/notebook/notebookRandomExecution.vscode.test.ts

```diff
@@ -0,0 +1,351 @@
+
+                    // Execution should have stopped.
+                    await waitForExecutionCompletedWithErrors(cell);
+                }
+            },
+            {
+                title: 'Test clearing when streaming output',
+                code: dedent`
+                            print("Start")
+                            import time
+                            for i in range(100):
+                                time.sleep(0.1)
+                                print(i)
+
+                            print("End")
+                            `,
+                assert: async (cell: NotebookCell) => {
+                    await waitForCondition(
+                        async () => kernelProvider.get(cell.notebook.uri) !== undefined,
+                        5_000,
+                        'No kernel'
+                    );
+                    await waitForCondition(async () => assertVSCCellIsRunning(cell), 15_000, 'Cell not being executed');
+                    await waitForTextOutputInVSCode(cell, 'Start', 0, false, 15_000); // Wait for 15 seconds for it to start (possibly
+
+                    // Wait till we get the desired output.
+                    await waitForCondition(
+                        async () =>
+                            assertHasTextOutputInVSCode(cell, 'Start', 0, false) &&
+                            assertHasTextOutputInVSCode(cell, '0', 0, false) &&
+                            assertHasTextOutputInVSCode(cell, '1', 0, false) &&
+                            assertHasTextOutputInVSCode(cell, '2', 0, false) &&
+                            assertHasTextOutputInVSCode(cell, '3', 0, false) &&
+                            assertHasTextOutputInVSCode(cell, '4', 0, false),
+                        15_000,
+                        'Cell did not get executed'
+                    );
+
+                    // Clear the cells
+                    await commands.executeCommand('notebook.clearAllCellsOutputs');
+
+                    // Wait till previous output gets cleared & we have new output.
+                    await waitForCondition(
+                        async () =>
+                            assertNotHasTextOutputInVSCode(cell, 'Start', 0, false) &&
+                            cell.outputs.length > 0 &&
+                            cell.outputs[0].outputKind === vscodeNotebookEnums.CellOutputKind.Rich,
+                        5_000,
+                        'Cell did not get cleared'
+                    );
+
+                    // Interrupt the kernel).
+                    await commands.executeCommand('notebook.cancelExecution');
+                    await waitForExecutionCompletedWithErrors(cell);
+
+                    // Verify that it hasn't got added (even after interrupting).
+                    assertNotHasTextOutputInVSCode(cell, 'Start', 0, false);
+                }
+            }
+        ];
+        return cells;
```

Comment:
**The goal here is to:**
* Create/edit cells with different content
* Run the cell or multiple cells
* Validate the output of the cells

Ensure the cells & state of the cells (execution state) is as expected.
We have had a lot of issues in the past & a few in the recent past where cell execution state is wrong & theres no way to recover from it.

**The plan here is log each operation so we can detect what combination causes the tests to fail**